zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -966,6 +966,117 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
   };
 
   var setTable = function(data, refresh) {
+    function addButtonMenuEvent(button, menu) {
```

Comment:
@corneadoug thanks. I will catch up later and learn how to better organize the code.